packages/@aws-cdk/aws-sagemaker/test/test.endpoint.ts

```diff
@@ -0,0 +1,170 @@
+                        initialVariantWeight: 20,
+                        instanceType: new ec2.InstanceType('p3.2xlarge'),
+                    }
+                ],
+                kmsKey,
+            });
+            endpoint.node.applyAspect(new cdk.Tag("Project", "myproject"));
+
+            // THEN
+            expect(stack).to(haveResource("AWS::SageMaker::EndpointConfig", {
+                EndpointConfigName: "myconfig",
+                ProductionVariants: [
+                    {
+                        InitialInstanceCount: 10,
+                        InitialVariantWeight: 20,
+                        InstanceType: 'ml.p3.2xlarge',
+                        ModelName: { "Fn::GetAtt": [ "Model2AD80A05", "ModelName" ]},
+                        VariantName: "myvariant",
+                    }
+                ],
+                KmsKeyId: { "Fn::GetAtt": [ "Key961B73FD", "Arn" ]  }
+            }));
+            expect(stack).to(haveResource("AWS::SageMaker::Endpoint", {
+                EndpointConfigName: { "Fn::GetAtt": [ "EndpointEndpointConfig5871F635", "EndpointConfigName" ] },
+                Tags: [
+                    { Key: "Name", Value: "Endpoint" },
+                    { Key: "Project", Value: "myproject" },
+                ]
+            }));
+
+            test.done();
+        },
+        "it throws error when no production variant given"(test: Test) {
+            // GIVEN
+            const stack = new cdk.Stack();
+
+            // WHEN
+            const endpoint = new sagemaker.Endpoint(stack, 'Endpoint');
+
+            const errors = validate(stack);
+
+            test.equal(errors.length, 1);
+            const error = errors[0];
+            test.same(error.source, endpoint);
+            test.equal(error.message, "Must have at least one Production Variant");
+
+            test.done();
+        },
+        "it throws error when incorrect instance count given"(test: Test) {
+            // GIVEN
+            const stack = new cdk.Stack();
+            const model = sagemaker.Model.fromModelName(stack, 'Model', "mymodel");
+
+            // WHEN
+            const endpoint = new sagemaker.Endpoint(stack, 'Endpoint', {
+                productionVariants: [
+                    {
+                        initialInstanceCount: -1,
+                        initialVariantWeight: 100,
+                        instanceType: ec2.InstanceType.of(ec2.InstanceClass.C4, ec2.InstanceSize.XLARGE),
+                        model,
+                        variantName: "production",
+                    }
+                ]
+            });
+            const errors = validate(stack);
+
+            test.equal(errors.length, 1);
+            const error = errors[0];
+            test.same(error.source, endpoint);
+            test.equal(error.message, "Must have at least one instance");
+
+            test.done();
+        },
+        "it throws error when incorrect variant weight given"(test: Test) {
```

Comment:
Please always have an empty line between tests on the same level of indentation.